core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -337,6 +338,8 @@ object KafkaConfig {
   val SslClientAuthProp = SslConfigs.SSL_CLIENT_AUTH_CONFIG
 
   /** ********* SASL Configuration ****************/
+  val SaslMechanismProp = SaslConfigs.SASL_MECHANISM
```

Comment:
The fact that it's common code is an implementation detail. We can pass the parameter via `ChannelBuilders` to deal with that. Let's see what @junrao thinks.